src/Vehicle/RemoteIDManager.cc

```diff
@@ -0,0 +1,424 @@
+                                                    _targetComponent,
+                                                    _id_or_mac_unknown,
+                                                    _settings->locationType()->rawValue().toUInt(),
+                                                    _settings->classificationType()->rawValue().toUInt(),
+                                                    geoPositionInfo.isValid() ? ( gcsPosition.latitude()  * 1.0e7 ) : 0, // If position not valid, send a 0
+                                                    geoPositionInfo.isValid() ? ( gcsPosition.longitude() * 1.0e7 ) : 0, // If position not valid, send a 0
+                                                    AREA_COUNT,
+                                                    AREA_RADIUS,
+                                                    -1000.0f,
+                                                    -1000.0f,
+                                                    _settings->categoryEU()->rawValue().toUInt(),
+                                                    _settings->classEU()->rawValue().toUInt(),
+                                                    geoPositionInfo.isValid() ? gcsPosition.altitude() : 0, // If position not valid, send a 0
+                                                    _timestamp2019()), // Time stamp needs to be since 00:00:00 1/1/2019
+        _vehicle->sendMessageOnLinkThreadSafe(sharedLink.get(), msg);
+    }
+}
+
+// Returns seconds elapsed since 00:00:00 1/1/2019
+uint32_t RemoteIDManager::_timestamp2019()
+{
+    uint32_t secsSinceEpoch2019 = 1546300800; // Secs elapsed since epoch to 1-1-2019
+
+    return ((QDateTime::currentDateTime().currentSecsSinceEpoch()) - secsSinceEpoch2019);
+}
+
+void RemoteIDManager::_sendBasicID()
+{
+    WeakLinkInterfacePtr weakLink = _vehicle->vehicleLinkManager()->primaryLink();
+
+    QString basicIDTemp = _settings->basicID()->rawValue().toString();
+    QByteArray ba = basicIDTemp.toLocal8Bit();
+
+    if (!weakLink.expired()) {
+        mavlink_message_t       msg;
+        SharedLinkInterfacePtr  sharedLink = weakLink.lock();
+
+        mavlink_msg_open_drone_id_basic_id_pack_chan(_mavlink->getSystemId(),
+                                                    _mavlink->getComponentId(),
+                                                    sharedLink->mavlinkChannel(),
+                                                    &msg,
+                                                    _targetSystem,
+                                                    _targetComponent,
+                                                    _id_or_mac_unknown,
+                                                    _settings->basicIDType()->rawValue().toUInt(),
+                                                    _settings->basicIDUaType()->rawValue().toUInt(),
+                                                    reinterpret_cast<const unsigned char*>(ba.constData())),
```

Comment:
Mavlink does not handle this as string, so you need to make sure the buffer is large enough.
You can use `ba.resize(MAVLINK_MSG_OPEN_DRONE_ID_BASIC_ID_FIELD_UAS_ID_LEN);`

Reply:
You are right, sorry I missed this one. I fixed it and left a comment about how this behaves ( adding padding bytes if array is smaller than size set, or discarding extra bytes if array is bigger than size set )